src/basic/verbs.c

```diff
@@ -79,17 +81,11 @@ int dispatch_verb(int argc, char *argv[], const Verb verbs[], void *userdata) {
         left = argc - optind;
         name = argv[optind];
 
-        for (i = 0;; i++) {
+        for (i = 0; verbs[i].dispatch; i++) {
                 bool found;
 
-                /* At the end of the list? */
-                if (!verbs[i].dispatch) {
-                        if (name)
-                                log_error("Unknown operation %s.", name);
-                        else
-                                log_error("Requires operation parameter.");
-                        return -EINVAL;
-                }
+                if (verbs[i].flags & VERB_FALLBACK)
+                        verb_fallback = &verbs[i];
```

Comment:
not sure I follow fully, but doesn't make the operation ambiguous? i.e. let's say you invoke "systemd-resolve quux" then this could either trigger the verb "quux" or the fallback with "quux" as argument?

Reply:
Yeah, I said this before, but this seems to be unchanged from the initial design:
the ambiguity created by the "host" verb being optional is not acceptable.

(I think "host" is also not a very good name, "query" would be better.)

Maybe instead of trying to shoe-horn all the existing functionality onto one command, it'd be better to provide two executables. They could even be the same binary, with the behaviour decided based on argv[0]:
- `systemd-resolve` would do the querying of names
- `resolvectl` would be mainly for querying and interaction of systemd-resolved,
- the functionality in `systemd-resolve` that is not covered by `resolvectl` would be deprecated, removed from `--help` and the man page, but still available for backward compatiblity.